nifi-nar-bundles/nifi-neo4j-bundle/nifi-neo4j-processors/src/main/java/org/apache/nifi/processors/neo4j/AbstractNeo4JCypherExecutor.java

```diff
@@ -0,0 +1,281 @@
+            .required(false)
+            .defaultValue(LOAD_BALANCING_STRATEGY_ROUND_ROBIN.getValue())
+            .allowableValues(LOAD_BALANCING_STRATEGY_ROUND_ROBIN, LOAD_BALANCING_STRATEGY_LEAST_CONNECTED)
+            .build();
+
+    public static final PropertyDescriptor CONNECTION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-time-out")
+            .displayName("Neo4J Max Connection Time Out (seconds)")
+            .description("The maximum time for establishing connection to the Neo4j")
+            .defaultValue("5 seconds")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_CONNECTION_POOL_SIZE = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-pool-size")
+            .displayName("Neo4J Max Connection Pool Size")
+            .description("The maximum connection pool size for Neo4j.")
+            .defaultValue("100")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_CONNECTION_ACQUISITION_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-acquisition-timeout")
+            .displayName("Neo4J Max Connection Acquisition Timeout")
+            .description("The maximum connection acquisition timeout.")
+            .defaultValue("60 second")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor IDLE_TIME_BEFORE_CONNECTION_TEST = new PropertyDescriptor.Builder()
+            .name("neo4j-idle-time-before-test")
+            .displayName("Neo4J Idle Time Before Connection Test")
+            .description("The idle time before connection test.")
+            .defaultValue("60 seconds")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor MAX_CONNECTION_LIFETIME = new PropertyDescriptor.Builder()
+            .name("neo4j-max-connection-lifetime")
+            .displayName("Neo4J Max Connection Lifetime")
+            .description("The maximum connection lifetime")
+            .defaultValue("3600 seconds")
+            .required(true)
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .sensitive(false)
+            .build();
+
+    public static final PropertyDescriptor ENCRYPTION = new PropertyDescriptor.Builder()
+            .name("neo4j-encryption")
+            .displayName("Neo4J Encrytion")
+            .description("Is connection encrypted")
```

Comment:
Can you please provide more detail on this in the description? Is this referring to TLS or a different encryption mechanism?

Reply:
@alopresto - Yes this is tls encryption as mentioned [here](https://neo4j.com/docs/developer-manual/current/drivers/client-applications/#driver-configuration-encryption)